components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java

```diff
@@ -4453,23 +4456,71 @@ public static String decryptToken(String token) throws CryptoException {
         if (Boolean.parseBoolean(config.getFirstProperty(APIConstants.ENCRYPT_TOKENS_ON_PERSISTENCE))) {
             return new String(CryptoUtil.getDefaultCryptoUtil().base64DecodeAndDecrypt(token), Charset.defaultCharset());
         }
+
+        if (Boolean.parseBoolean(config.getFirstProperty(APIConstants.HASH_TOKENS_ON_PERSISTENCE))) {
+            return null;
+        }
         return token;
     }
 
     /**
      * @param token
      * @return
      */
-    public static String encryptToken(String token) throws CryptoException {
+    public static String encryptToken(String token) throws CryptoException, APIManagementException {
         APIManagerConfiguration config = ServiceReferenceHolder.getInstance().
                 getAPIManagerConfigurationService().getAPIManagerConfiguration();
 
         if (Boolean.parseBoolean(config.getFirstProperty(APIConstants.ENCRYPT_TOKENS_ON_PERSISTENCE))) {
             return CryptoUtil.getDefaultCryptoUtil().encryptAndBase64Encode(token.getBytes(Charset.defaultCharset()));
         }
+
+        if (Boolean.parseBoolean(config.getFirstProperty(APIConstants.HASH_TOKENS_ON_PERSISTENCE))) {
+            try {
+                return hash(token);
+            } catch (IdentityOAuth2Exception e) {
+                throw new APIManagementException("Error while hashing the token.");
+            }
+        }
         return token;
     }
 
+    /**
+     * Method to generate hash value.
+     *
+     * @param plainText Plain text value.
+     * @return hashed value.
+     */
+    private static String hash(String plainText) throws IdentityOAuth2Exception {
```

Comment:
why this method is private

Reply:
This method is used only within this class